types/pipeline_status.go

```diff
@@ -2,11 +2,27 @@ package types
 
 import "time"
 
+type PipelineEventSource string
+
+const (
+	PipelineEventSourceDeployment PipelineEventSource = "k8s:deployment"
+	PipelineEventSourcePod        PipelineEventSource = "k8s:pod"
+)
+
+// PipelineEvent model.
+type PipelineEvent struct {
```

Comment:
Don't you want to add the `Action` field too? Might be useful. `Source` too.

https://pkg.go.dev/k8s.io/api/core/v1#Event

Reply:
I think it contains the http verb that caused the change, which is not that useful to the end user.

Reply:
I don't think action is any useful here.